library/Translate/src/Juvix/Contextify.hs

```diff
@@ -37,14 +37,14 @@ op names = do
   case context of
     Left err -> pure $ Left err
     Right ctx -> do
-      (newCtx, _) <- Environment.runMIO (Passes.resolveModule ctx)
+      (newCtx, _) <- -- only Passes.resolveModule requires IO in these passes
+        Environment.runMIO $
+          Passes.resolveModule ctx
+            >>= Passes.inifixSoloPass
+            >>= Passes.recordPi
       case newCtx of
         Left err -> pure $ Left $ PassErr err
-        Right t ->
-          let (infix', _) = Environment.runM (Passes.inifixSoloPass t)
-           in case infix' of
-                Left err -> pure $ Left $ PassErr err
-                Right x -> pure (Right x)
+        Right t -> pure (Right t)
```

Comment:
Wonder if we could have in out standard library a function `left` and `right` 
```
left :: (a -> c) -> Either a b -> Either c b
left = flip bimap identity
```

so that these cases
```
      case newCtx of
        Left err -> pure $ Left $ PassErr err
        Right t -> pure (Right t)
```
become
```
      pure $ left PassErr newCtx
```

Reply:
Maybe 
```
mapLeft :: (a -> c) -> Either a b -> Either c b
mapLeft = flip bimap identity
```
and
```
left :: (a -> c) -> Either a b -> c
left = flip either identity
```
(same on the right)